test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -11110,13 +11110,15 @@ def add_server_dyn_res(self, custom_resource, script_body=None,
         else:
             if dirname is None:
                 dirname = self.pbs_conf['PBS_HOME']
-            res_file = self.du.create_temp_file(prefix=prefix, suffix=suffix,
+            tmp_file = self.du.create_temp_file(prefix=prefix, suffix=suffix,
                                                 body=script_body,
-                                                asuser="root",
-                                                dirname=dirname, hostname=host,
-                                                mode=perm)
+                                                hostname=host)
+            res_file = os.path.join(dirname, tmp_file.split(os.path.sep)[-1])
+            self.du.run_copy(self.hostname, tmp_file, res_file, sudo=True,
+                             preserve_permission=False)
+            self.du.chown(path=res_file, uid=0, gid=0, sudo=True)
+            self.du.chmod(self.hostname, path=res_file, mode=perm, sudo=True)
             self.dyn_created_files.append(res_file)
```

Comment:
As per our offline discussion, I thought you were going to revert this since @latha-subramanian is implementing this for all files created by create_temp_file()